geospatial/src/Dem.cc

```diff
@@ -161,12 +176,38 @@ int Dem::Load(const std::string &_filename)
   if (this->LoadData() != 0)
     return -1;
 
-  // Set the min/max heights
-  this->dataPtr->minElevation = *std::min_element(&this->dataPtr->demData[0],
-      &this->dataPtr->demData[0] + this->dataPtr->side * this->dataPtr->side);
-  this->dataPtr->maxElevation = *std::max_element(&this->dataPtr->demData[0],
-      &this->dataPtr->demData[0] + this->dataPtr->side * this->dataPtr->side);
+  // Check for nodata value in dem data. This is used when computing the
+  // min elevation. If nodata value is not defined, we assume it will be one
+  // of the commonly used values such as -9999, -32768, etc.
+  // See https://desktop.arcgis.com/en/arcmap/10.8/manage-data/raster-and-images/nodata-in-raster-datasets.htm
+  // For simplicity, we will treat values <= -9999 as nodata values and
+  // ignore them when computing the min elevation.
+  int validNoData = 0;
+  const double defaultNoDataValue = -9999;
+  double noDataValue = this->dataPtr->band->GetNoDataValue(&validNoData);
+  if (validNoData <= 0)
+    noDataValue = defaultNoDataValue;
+
+  double min = ignition::math::MAX_D;
+  double max = -ignition::math::MAX_D;
+  for (auto d : this->dataPtr->demData)
+  {
+    if (d > noDataValue)
```

Comment:
While testing this PR with a DEM that uses NaN as the no data value, I noticed that this logic doesn't handle NaNs well, because any comparison to NaN will always return false.

How about something like this:

```cpp
  for (auto d : this->dataPtr->demData)
  {
    // All comparisons to NaN return false, so guard against NaN NoData
    if (!std::isnan(noDataValue) && d <= noDataValue)
      continue;

    if (!std::isfinite(d))
      continue;

    if (d < min)
      min = d;
    if (d > max)
      max = d;
  }
```